### DISCUSSION TITLE
AttributeError: 'NoneType' object has no attribute 'total_seconds' while using util.df

### DISCUSSION THREAD

**Discussion by GorkyU** [erdewit/ib_insync#609] [General]
Hi guys,

Whenever I try using util.df to a BarData, I am receving the error below

AttributeError: 'NoneType' object has no attribute 'total_seconds'

![image](https://github.com/erdewit/ib_insync/assets/137977147/dfb7f00b-145a-42c1-97bf-b5b4719d0aed)

Any idea about what could be the reason?

Thanks

**Comment by smelnich48** [comment]
I am using this workaround:

  
```python
df = util.df(bars)
df["date"] = df["date"].dt.tz_convert("America/New_York")
print(df)
```   

        
        It seems that the df does not know what timezone to use.

**Comment by meyerholsen** [comment]
I edited  function parseIBDatetime() in util.py file by adding   t=pd.to_datetime(t.isoformat())    after string number 544. Now it works fine. The mistake appeared because of updates in Python 3.9
I found this solution in the internet, here:
https://groups.io/g/insync/topic/util/95764816?p=,,,20,0,0,0::recentpostdate/sticky,,,20,2,20,95764816,previd%3D1672752190301790742,nextid%3D1671183202943560474&previd=1672752190301790742&nextid=1671183202943560474
